public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -23,20 +23,20 @@ export const PieChartPanel: React.FC<Props> = ({
     [fieldConfig, onFieldConfigChange]
   );
 
-  const values = getFieldDisplayValues({
+  const fieldDisplayValues = getFieldDisplayValues({
     fieldConfig,
     reduceOptions: options.reduceOptions,
     data: data.series,
     theme: useTheme(),
     replaceVariables: replaceVariables,
     timeZone,
-  }).map((v) => v.display);
+  });
 
   return (
     <PieChart
       width={width}
       height={height}
-      values={values}
+      fieldDisplayValues={fieldDisplayValues}
```

Comment:
Not directly related to data links, but maybe we could follow one direction when it comes to panel -> visualization data flow? Instead of passing fieldDisplayValues I would expect PieChart to accept dataFrame(s) and figure out the field display value on it's own. Then panel's responsibility os only to manipulate the options and the visualiation handles iz specific operations?

WDYT @torkelo ?

Reply:
yea, that sounds like a good change, should make it easier to re-use in other contexts

Reply:
added that to https://github.com/grafana/grafana/issues/31116